### PR TITLE
Silence `unused_imports` for redundant imports

### PR DESCRIPTION
**Pull request by compiler-errors**
Quick fix for https://github.com/rust-lang/rust/issues/121708#issuecomment-2048105393

r? @petrochenkov cc @joshtriplett 

I think this is right, would like confirmation. I also think it's weird that we're using `=` to assign to `is_redundant` but using `per_ns` for the actual spans. Seems like this could be weirdly order dependent, but that's unrelated to this change.

### CODE REVIEW

File: compiler/rustc_resolve/src/imports.rs [hunk 1]
```diff
@@ -1375,8 +1375,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     target_bindings[ns].get(),
                 ) {
                     Ok(other_binding) => {
-                        is_redundant =
-                            binding.res() == other_binding.res() && !other_binding.is_ambiguity();
+                        is_redundant = binding.res() == other_binding.res()
```
Comment (compiler-errors) [inline compiler/rustc_resolve/src/imports.rs:1378]: Or should we just suppress all redundant imports? 🤔 

That would be easier to throw under a new lint name if so.

**Comment by petrochenkov** [comment]
>Or should we just suppress all redundant imports?

Yes, this PR as is doesn't cover redundant stdlib prelude imports, which are the largest source of regressions (they don't look like globs from other crates).

I suggest just commenting out the `buffer_lint_with_diagnostic` call in `fn check_for_redundant_imports` on beta.
And migrate the whole logic to a new `redundant_imports` lint on master.
@rustbot author

**Comment by compiler-errors** [comment]
@rustbot ready

Ok, just silenced the whole thing

**Comment by compiler-errors** [comment]
🤦 push didn't go through, one sec

File: tests/ui/lint/use-redundant/use-redundant-prelude-rust-2015.rs [hunk 2]
```diff
@@ -2,11 +2,15 @@
 #![warn(unused_imports)]
 
 
-use std::option::Option::Some;//~ WARNING the item `Some` is imported redundantly
-use std::option::Option::None; //~ WARNING the item `None` is imported redundantly
+use std::option::Option::Some;
+//FIXME(unused_imports): ~^ WARNING the item `Some` is imported redundantly
```
Comment (compiler-errors) [inline tests/ui/lint/use-redundant/use-redundant-prelude-rust-2015.rs:6]: Line break b/c line was too long; alternatively could throw out these altogether, but it's nice to keep em around imo.

**Comment by petrochenkov** [comment]
Ok, it's probably fine to land this on master as well.
@bors r+

**Comment by apiraino** [comment]
Beta backport approved as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-04-18/near/434055620). A backport PR will be authored by the release team at the end of the current development cycle.

@rustbot label +beta-accepted

**Pull request merged**